### PR TITLE
Fix several compiler warnings

### PR DESCRIPTION
**Pull request by wwmayer**
* -Wmaybe-uninitialized
* -Wunused-parameter
* -Wunused-variable
* -Wnonnull
* -Wstringop-truncation
* -Wstringop-overflow

### CODE REVIEW

File: src/Mod/Sketcher/App/SketchObject.cpp [hunk 1]
```diff
@@ -9478,6 +9478,7 @@ std::pair<std::string,std::string> SketchObject::getElementName(
     index.appendToStringBuffer(ret.second);
     if (auto realName = convertInternalName(ret.second.c_str())) {
         Data::MappedElement mappedElement;
+        (void)realName;
```
Comment (chennes) [inline src/Mod/Sketcher/App/SketchObject.cpp:9481]: Why did you decide not to use `boost::ignore_unused` here as well? (It's new to me, I've never used it before)

Comment (wwmayer) [reply src/Mod/Sketcher/App/SketchObject.cpp:9481]: Because below there is commented some code which I guess will be re-activated in the future.

**Comment by chennes** [comment]
Do you know the clang equivalents for stringop-truncation and stringop-overflow?

**Comment by wwmayer** [comment]
No, I don't know. But when looking at https://clang.llvm.org/docs/DiagnosticsReference.html then there is a https://clang.llvm.org/docs/DiagnosticsReference.html#wstrncat-size  which looks quite similar.

**Pull request merged**